### ISSUE TITLE
Consider using Derive for X.509 encoding

### ISSUE THREAD

**Issue by jhand2** [chipsalliance/caliptra-dpe#290]
The X.509 encoding library is a bit hard to follow/contribute to. It might be worth making it more like the asn1 library, where you derive an encode trait on a struct that holds the data to be encoded.

It might even be worth trying to use the asn1 crate and checking how much size it adds to the binary.

Not high priority, just a thought.

**Comment by clundin25** [comment]
We tried but the code size was too much, even on the 2.x platform.
